### DISCUSSION TITLE
Auto-cleaning unused deployments due to no server storage space left

### DISCUSSION THREAD

**Discussion by kroodev** [coollabsio/coolify#3684] [Q&A] [answered]
After a month of stable committing, I found that recent deployments were failing due to no space left on the server. While I found some mentions of it in the discussion, I am not sure I found a solid solution. 

So I ran `docker system df`, found out that I had 200GB of Docker images, and then ran `docker images` to see all of them. 
To remove all unused Docker images, I ran `docker image prune -a`.

I wonder how other people solve this problem of cleaning up the server. I didn't find a setting to auto-remove old deployments in Coolify, so I am considering adding a cron task to remove unused Docker images daily. 

- Did you find any better solution to cleaning up the server's space?
- Is it even safe to run `docker image prune -a`?

Edit: I use Nixpacks if that matters.

**Comment by peaklabs-dev** [comment, marked as answer]
This is already availabe just turn on force cleanup in the server settings. Check to helper text (i) in Coolify to see what gets cleaned up. And yes it is safe to remove unused images they are non persistent and can just be re-built or re-downloaded

**Reply by kroodev** [reply]
Found this setting! Thank you!
